Modules/sha1module.c

```diff
@@ -71,6 +71,9 @@ static SHA1object *
 newSHA1object(SHA1State *st)
 {
     SHA1object *sha = (SHA1object *)PyObject_GC_New(SHA1object, st->sha1_type);
+    if (sha == NULL) {
+        return NULL;
```

Comment:
I think it's required to call `PyErr_NoMemory();` here.

Reply:
Not really. Unlike `PyMem_*` functions, both `PyObject_GC_New` and `PyObject_New` set `MemoryError` if fail to allocate memory for object.